examples/gno.land/r/system/validators/public.gno

```diff
@@ -0,0 +1,60 @@
+package validators
+
+import (
+	"std"
+)
+
+func CreateValidator(name, pubkey string) {
+	std.AssertOriginCall()
+
+	if validators.Size() >= maxValidators {
+		panic("validator set is full")
+	}
+
+	caller := std.GetOrigCaller()
+	vkey := validatorKey(caller)
+	if validators.Has(vkey) {
+		panic("validator already registered")
+	}
+
+	power := std.GetOrigSend().AmountOf(nativeDenom)
```

Comment:
I would keep the voting power completely from this contract, and would just make CreateValidator checking for `assertIsAdmin`.

Reply:
I will change it.  Is it possible to make `CreateValidator` admin-only (with voting power)? :-)

Reply:
To keep things simple, I changed it so that anyone can register a node with `CreateValidator()` and the `admin` can set the `votingpower` with `UpdateValidatorPower()`.